moveit_core/collision_detection_fcl/src/collision_common.cpp

```diff
@@ -474,45 +468,37 @@ bool distanceCallback(fcl::CollisionObject* o1, fcl::CollisionObject* o2, void*
     CONSOLE_BRIDGE_logDebug("Actually checking collisions between %s and %s", cd1->getID().c_str(),
                             cd2->getID().c_str());
 
+  fcl::DistanceResult fcl_result;
   DistanceResultsData dist_result;
   double dist_threshold = cdata->req->distance_threshold;
-  std::map<std::string, DistanceResultsData>::iterator it1, it2;
 
-  if (!cdata->req->global_minimum_only)
-  {
-    it1 = cdata->res->distances.find(cd1->ptr.obj->id_);
-    it2 = cdata->res->distances.find(cd2->ptr.obj->id_);
+  const std::pair<std::string, std::string>& pc = cd1->getID() < cd2->getID() ?
+                                                        std::make_pair(cd1->getID(), cd2->getID()) :
+                                                        std::make_pair(cd2->getID(), cd1->getID());
+
+  DistanceMap::iterator it = cdata->res->distances.find(pc);
 
-    if (cdata->req->active_components_only)
+
+  if (it != cdata->res->distances.end())
```

Comment:
all this logic you've modified could really use some explanation comments

Reply:
Before the latest commit it was only possible to get the global minimum distance or the minimum distance for a given link. This logic change adds the ability to get all of the contacts for a given pair of links. This is useful if a link is composed of multiple geometries and by getting all of the contacts you can accurately define a gradient for those pair of links to move them out of collision.